fairscale/experimental/nn/offload.py

```diff
@@ -388,21 +311,28 @@ class OffloadModel(nn.Module):
 
     def __init__(
         self,
-        model_cpu: nn.Sequential,
+        model: nn.Sequential,
         device: torch.device,
         offload_device: torch.device = torch.device("cpu"),
-        num_slices: int = 5,
+        num_slices: int = 3,
         checkpoint_activation: bool = False,
         num_microbatches: int = 1,
     ):
         super().__init__()
-        # TODO(anj-s): Add error checks for cuda and sequential model.
+        assert model
+        assert device
```

Comment:
if these are error checking, it is ideal not to use assert since python -O can bypass asserts. It is better to raise errors.